app/src/main/kotlin/com/wire/android/navigation/NavigationManager.kt

```diff
@@ -21,18 +21,29 @@
 package com.wire.android.navigation
 
 import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.take
 
 class NavigationManager {
 
-    var navigateState = MutableSharedFlow<NavigationCommand?>()
-    var navigateBack = MutableSharedFlow<Map<String, Any>>()
+    private val _navigateState = MutableSharedFlow<NavigationCommand?>()
+    private val _navigateBack = MutableSharedFlow<Map<String, Any>>()
+    var navigateState: SharedFlow<NavigationCommand?> = _navigateState
+    var navigateBack: SharedFlow<Map<String, Any>> = _navigateBack
 
     suspend fun navigate(command: NavigationCommand) {
-        navigateState.emit(command)
+        // in case of DeepLink possible scenario when navigate() is called, but _navigateState Flow is not subscribed yet,
+        // so the navigate command goes nowhere.
+        // To avoid such lose we'll wait till _navigateState Flow is subscribed and emit command into it only after that.
```

Comment:
Wouldn't be better to set a `replay` parameter to a `MutableSharedFlow` so we could emit items before it's subscribed and when subscribed it would get the latest emitted value?

Reply:
`replay` is deprecated and `shareIn` requires a CoroutineContext

Reply:
Hmm, I don't see `replay` being deprecated and we use it this way in `IncrementalSyncRepository` for instance, maybe you are thinking about something other than a constructor parameter? 😄 

<img width="827" alt="image" src="https://user-images.githubusercontent.com/30429749/223435036-04e6497b-68c9-4ffc-81de-fdb5419ccc35.png">